### ISSUE TITLE
No SC2086 warning when double quotes are required to prevent word splitting.

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1024]
#### For bugs
- Rule Id: `SC2086`
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

if [ "${DEBUG:=0}" != 0 ]; then
  echo 'Debug...'
fi

if [ $DEBUG = 0 ]; then
  echo 'No debug...'
fi

```

#### Here's what shellcheck currently says:

Nothing.

#### Here's what I wanted or expected to see:

```
Line 7:
if [ $DEBUG = 0 ]; then
     ^-- SC2086: Double quote to prevent globbing and word splitting.
```
Debug could be set to a variable with spaces and it will then fail when that is true so `$DEBUG` should be double quoted to prevent word splitting.
```
$ DEBUG='foo bar' ./test.sh 
Debug...
./test.sh: line 7: [: too many arguments
```
Alternatively this example should work without double quotes as `$DEBUG` can only be `0` or `1` and should continue to not show any warnings.
```
#!/bin/sh

[ "${DEBUG:=0}" != 0 ] && DEBUG=1

if [ $DEBUG = 0 ]; then
  echo 'No debug...'
fi
```

**Comment by nihilus** [comment]
Ive been hit by gris as well, I guess it only and wrongly catches the first occurrence of the variable.
